odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1701,7 +1701,7 @@ def _validate_tag_button(self, node, name_manager, node_info):
             if type_ != 'action' and type_ != 'object':
                 return
             elif not name:
-                self._raise_view_error(_("Button must have a name"), node)
+                return
```

Comment:
I'm not sure you have to remove this validation (and I would like to keep it).

If I understand correctly, it's for the studio flow, where the user adds a button, but before he can specify the method, the view is already validated. A workaround would be to wait for the name to be set to set the type attribute. What do you think?

Reply:
> I'm not sure you have to remove this validation (and I would like to keep it).

We discussed with CTH and we thought this can be removed.
Yes, that was the main reason. We wanted to simplify the studio editor flow when adding a button **without having a dialog to set all attributes at once** and also for **changing its type without having to remove the button**.

I don't know how and what we could do in studio to avoid having the view validation when the name attribute is not set.
It is not just when the button is added, but also if the user clears the name value, or change its type (the name attribute is cleared)

Reply:
I don't think it's difficult: the name property in the sidebar should "simply" set/unset two attributes: the name, and the type.

Reply:
Studio saves each operation one after the other. For each change occuring in the sidebar. There is no "edit mode and save after it's ready"

Reply:
Ok let's move forward